src/Job/Job.ts

```diff
@@ -48,6 +48,13 @@ export interface Job {
 	 */
 	getPrerequisiteFilesIterable(): Iterable<string>
 
+	/**
+	 * Gets this Job's dependencies (its prerequisites, recursively)
+	 *
+	 * @returns an array containing a deep scan of the Job's prerequisites.
+	 */
+	getDeepPrerequisitesIterable(): Array<Job>
```

Comment:
Why not make this return a `Set`? It seems like order doesn't matter.

Reply:
And if you're going to make it return a `Set`, it'll probably be easiest to just return an `Iterable` like the other functions.

Reply:
I think order matters, because the way it is right now, pre-req jobs occur before the jobs that require them in the array. If this behavior has no function when it coming to tarball generation later on, I think we can change it